.github/workflows/code_format_checks.yml

```diff
@@ -0,0 +1,21 @@
+name: Code Format Checks
+on: [push, pull_request]
+jobs:
+  code_format_checks:
+    strategy:
+      matrix:
+        go: ['1.16']
+        platform: [ubuntu-latest]
+    runs-on: ${{ matrix.platform }}
+    steps:
+      - name: Install Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ matrix.go }}
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Run Code Format Checks
+        run: |
+          make test-fmt &&
```

Comment:
there was the subtree check too right? I think this test that release-tools wasn't modified and that I think is important since this folder comes from a different repo.

could we run `make all` as it is in Linux instead?

Reply:
`test-subtree` was failing in my previous attempt to refactor the GitHub actions: PR #261. In fact I think this issue would have been happening all along, but the error was not showing because `make all` fails on `test-fmt` and never got to execute `test-subtree`. 

The reason behind this is that GitHub Action's checkout@v2 grabs the latest commit as a graft, which causes the `git log -- release-tools` command run by verify-subtree to think that the last commit added the entire repository (or something like that).

I did some digging, and it seems like no other repo in CSI Proxy runs verify-subtree in a GitHub action. I think we can just remove it and run `make test-fmt test-vendor test-shellcheck`.